docs/modules/ROOT/pages/components.adoc

```diff
@@ -0,0 +1,186 @@
+= Using Components
+
+The following documentation provides reasoning and examples on how to use Contracts for Cairo components.
+
+== Components
+
+:shamans-post: https://community.starknet.io/t/cairo-components/101136#components-1[Starknet Shamans post]
+
+Starknet components are separate modules that contain storage, events, and implementations that can be integrated into a contract.
+Components themselves cannot be declared or deployed.
+Another way to think of components is that they are abstract modules that must be instantiated.
+
+TIP: For more information on the construction and design of Starknet components, see the {shamans-post}.
+
+=== Setting up
+
+:initializable-component: xref:/security.adoc#initializable[InitializableComponent]
+
+Integrating a component first requires that the contract declares the component with the `component!` macro:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    // Import the component
+    use openzeppelin::security::InitializableComponent;
+
+    // Declare the component
+    component!(path: InitializableComponent, storage: initializable, event: InitializableEvent);
+}
+----
+
+The `path` argument should be the imported component itself (in this case, {initializable-component}).
+Notice that the `storage` and `event` arguments are representations set within the macro.
+In other words, the `initializable` and `InitializableEvent` names follow the Contracts for Cairo convention, but they can be renamed.
+
+=== Accessing component state
+
+Before a contract can access a component's state, the contract must integrate the implementations that will be used.
```

Comment:
Why is the focus put on accessing state? Isn't thi section more about importing impls? That way it's also a good segue to embedding them